src/parser/bgp/messages.rs

```diff
@@ -45,30 +45,22 @@ pub fn parse_bgp_message(
     have the smallest value required, given the rest of the
     message.
     */
-    let length = data.get_u16();
+    let length = data.read_u16()?;
     if !(19..=4096).contains(&length) {
-        return Err(ParserError::ParseError(format!(
-            "invalid BGP message length {}",
-            length
-        )));
+        return Err(ParserError::InvalidBgpMessageLength(length));
     }
 
+    // TODO: Why do we sometimes change our length estimate?
```

Comment:
Need more context for this question.

Reply:
My confusion comes from this piece of code. Shouldn't the total size always greater or equal to the length of the message?
```rust
let bgp_msg_length = if (length as usize) > total_size {
    total_size - 19
} else {
    length as usize - 19
};
```
Could this be related to partial attributes being cutoff?

Reply:
There is no guarantee that the number of bytes available matches the number of bytes encoded in MRT message.

`length` is read from the MRT message:
```
let length = data.read_u16()?;
```
It is completely possible that the value does not match the MRT message size we actually have. Therefore we have that check here to identify potential data issues. Here as we have already read all the bytes of the given MRT record into message, the length mismatch does not impact the parsing of the following records, and thus we made warnings instead of hard errors in the current version.